File: python/leetcode/057.py
class Solution(object):
    def insert(self, intervals, newInterval):
        start, end = newInterval.start, newInterval.end
        left = [item for item in intervals if item.end < start]
        right = [item for item in intervals if item.start > end]
        if left + right != intervals:
            start = min(start, intervals[len(left)].start)
            end = max(end, intervals[~len(right)].end)  
            # ~len(right) = - len(right) - 1
        return left + [Interval(start, end)] + right

File: cpp/leetcode/057.cpp
class Solution {
public:
    vector<Interval> insert(vector<Interval>& intervals, Interval newInterval) {
        vector<Interval> res;
        int i = 0;
        for(; i < intervals.size(); ++i){
            if(intervals[i].end < newInterval.start) {
                res.push_back(intervals[i]);
            } else if(intervals[i].start > newInterval.end) {
                break;
            } else {
                newInterval.start = min(newInterval.start, intervals[i].start);
                newInterval.end = max(newInterval.end, intervals[i].end);
            }
        }
        res.push_back(newInterval);
        for(;i < intervals.size(); ++i) {
            res.push_back(intervals[i]);

        return res;
    }
};